routes/doctorRouter.js

```diff
@@ -1,13 +1,18 @@
 const express = require('express');
 const path = require('path');
 
-const { advancedResults, fileUpload } = require('../middleware');
+const { advancedResults, fileUpload, authorize } = require('../middleware');
 const { doctorController, doctorAuthentication } = require('../controllers');
+const { authFun } = require('../utils');
 
 const router = express.Router();
```

Comment:
shouldn't you use here options object with mergeParams passed? As you've redirected from parent router (User) to a child router (Doctor). Like this

```js
const router = express.Router({ mergeParams: true });
```